### PR TITLE
fix: get tests passing on plucky

### PR DESCRIPTION
**Pull request by Perfect5th**
Twisted changed a little bit - it used to run `cleanFailure` on `Failure` instances in the `defer` code but no longer does in the version available in plucky (python3-twisted 24.10).

This change does `cleanFailure` in the `FakeReactor` we use for some tests. This allows tests that rely on catching and flushing `Failure`s to succeed.

### CODE REVIEW

**Comment by Perfect5th** [comment]
One way to test, created a plucky container using `dev_lxc` (https://pypi.org/project/dev-lxc/) and run the tests, with and without this change:
```
dev_lxc create --config .dev-lxc/config.yaml plucky
dev_lxc shell plucky
make check
```

**Review by wck0 — APPROVED** [review]
Tests passed on `plucky`, `noble`, `jammy`, and `focal`.

One in-line comment asking to review some grammar.

Also, codecov seems confused. Not sure how to address that...

File: landscape/lib/testing.py [hunk 1]
```diff
@@ -685,7 +685,15 @@ def fake():
...
             try:
-                f(*args, **kwargs)
+                deferred = f(*args, **kwargs)
+
+                if (
+                    hasattr(deferred, "result")
+                    and isinstance(deferred.result, Failure)
+                ):
+                    # Required for some failures to get GC'd properly flushed.
+                    # Twisted did this in versions < 24.10, but now we have to.
```
[the remark this answers is not in the document]
Comment (Perfect5th) [reply landscape/lib/testing.py:695]: Oh yeah, thanks. Will fix.

**Comment by Perfect5th** [comment]
I'm not _too_ worried about codecov. If anything, we should disable it in favour of TiCS.

**Pull request merged**